gobblin-service/src/main/java/org/apache/gobblin/service/modules/dataset/DatasetDescriptor.java

```diff
@@ -17,28 +17,54 @@
 
 package org.apache.gobblin.service.modules.dataset;
 
+import com.typesafe.config.Config;
+
 import org.apache.gobblin.annotation.Alpha;
 
 
 /**
- * The interface for dataset descriptors.
+ * The interface for dataset descriptors. Each dataset is described in terms of the following attributes:
+ *  <ul>
+ *    <p> platform (e.g. HDFS, ADLS, JDBC). </p>
+ *    <p> path, which describes the fully qualified name of the dataset. </p>
+ *    <p> a format descriptor, which encapsulates its representation (e.g. avro, csv), codec (e.g. gzip, deflate), and
+ *    encryption config (e.g. aes_rotating, gpg). </p>
+ *  </ul>
  */
 @Alpha
 public interface DatasetDescriptor {
   /**
-   * @return the dataset platform i.e. the storage backing the dataset (e.g. HDFS, JDBC, Espresso etc.)
+   * @return the dataset platform i.e. the storage system backing the dataset (e.g. HDFS, ADLS, JDBC etc.)
    */
   public String getPlatform();
 
+  /**
+   * Returns the fully qualified name of a dataset. The fully qualified name is the absolute directory path of a dataset
+   * when the dataset is backed by a FileSystem. In the case of a database table, it is dbName.tableName.
+   * @return dataset path.
+   */
+  public String getPath();
+
+  /**
+   *
+   * @return storage format of the dataset.
+   */
+  public FormatConfig getFormatConfig();
+
   /**
    * @return a human-readable description of the dataset.
    */
   public String getDescription();
 
   /**
-   * @return true if this {@link DatasetDescriptor} is compatible with the other {@link DatasetDescriptor} i.e. the
-   * datasets described by this {@link DatasetDescriptor} is a subset of the datasets described by the other {@link DatasetDescriptor}.
-   * This check is non-commutative.
+   * @return true if this {@link DatasetDescriptor} contains the other {@link DatasetDescriptor} i.e. the
+   * datasets described by this {@link DatasetDescriptor} is a subset of the datasets described by the other
+   * {@link DatasetDescriptor}. This operation is non-commutative.
+   */
+  public boolean contains(DatasetDescriptor other);
```

Comment:
I prefer to leave it as contains(), since it captures the set relationship. At any rate, isApplicableTo() is the same as isCompatibleWith() that I had earlier.